server/db/s3.js

```diff
@@ -151,7 +153,31 @@ module.exports = {
 
   /* UPDATE operations */
 
-  // TODO: support incremental updates (#92)
+  /**
+   * Append data to an existing schema entry.
+   * @param {string} schemaName the schema name
+   * @param {string} userID the userID
+   * @param {string} newData the data to append
+   * @param {s3Callback} callback the callback that handles the AWS response
+   */
+  appendEntry: function(schemaName, userID, newData, callback) {
+    var bucketName = getBucketNameForSchemaName(schemaName);
+    var params = {
+      Bucket: bucketName,
+      Key: userID
+    };
+    // Since AWS S3 doesn't natively support append operations
+    // (see https://forums.aws.amazon.com/message.jspa?messageID=540395),
+    // need to get the existing object, append to it, and upload the result.
+    s3.getObject(params, function(error, entry) {
+      if (error) {
+        callback(error);
+      } else {
+        var data = Buffer.concat([entry.Body, newData]);
```

Comment:
Will this combine the strings of the two files? I'm worried this will mess up .csv files being stored. csv files store their headers in the first row and if we append two together there will be a list of the headers in the middle.

For example:

thing1.csv - 

name, animal, age
murphy, dog, 7
thomas, cat, 3
bilbo, cat, 19

thing2.csv - 

name, animal, age
glump, cat, 2
blimbo, dog, 10
glandolf, cat, 1

when combined will become:
name, animal, age
murphy, dog, 7
thomas, cat, 3
bilbo, cat, 19
name, animal, age
glump, cat, 2
blimbo, dog, 10
glandolf, cat, 1